src/internals/scan.rs

```diff
@@ -64,23 +65,28 @@ pub fn rules_scan_mem<'a>(
     mem: &[u8],
     timeout: i32,
     flags: i32,
-) -> Result<Vec<Rule<'a>>, YaraError> {
-    let mut results = Vec::<Rule<'a>>::new();
+    callback: impl FnMut(CallbackMsg<'a>) -> CallbackReturn,
+) -> Result<(), YaraError> {
+    let p_callback: Box<Box<dyn FnMut(CallbackMsg<'a>) -> CallbackReturn>> =
+        Box::new(Box::new(callback));
+    let user_data = Box::into_raw(p_callback) as *mut c_void;
     let result = unsafe {
         yara_sys::yr_rules_scan_mem(
             rules,
             mem.as_ptr(),
             mem.len().try_into().unwrap(),
             flags,
             Some(scan_callback),
-            &mut results as *mut Vec<_> as *mut c_void,
+            user_data,
```

Comment:
Can't you just use something like `&*p_callback as &mut _` ? So you don't have to leak the pointer and re-create a box after?

Reply:
Yes, it work! Thank you!